dmake/build.py

```diff
@@ -102,31 +103,33 @@ def build(self):
     def tag(self):
         template_kwargs = template_args.tag_template_args()
         for push_mode, repo, tag_template in self.pushes:
+            # continue to next item if not needed
             need_push = self.need_push(push_mode)
+            if not need_push:
```

Comment:
I think there is no need to bypass tagging once `need_push` being `False`,  since tagging is just a pre-step of a push,  and tagging alone will not have any side-effect on the world,  and tags will always simplify the management of local images.